airbyte-integrations/connectors/source-mongodb/lib/mongodb_reader.rb

```diff
@@ -0,0 +1,81 @@
+require_relative './airbyte_protocol.rb'
+require_relative './airbyte_logger.rb'
+
+require_relative './mongodb_stream.rb'
+
+class MongodbReader
+  BATCH_SIZE = 10_000
+  LOG_BATCH_SIZE = 10_000
+
+  def initialize(client:, catalog:, state:)
+    @client = client
+    @catalog = catalog
+    @state = state
+  end
+
+  def read
+    @catalog['streams'].each do |stream_object|
+      stream = stream_object['stream']
+      AirbyteLogger.log("Reading stream #{stream['name']} in #{stream['sync_mode']} mode")
```

Comment:
`sync_mode` should be a field on `stream_object`. See [`ConfiguredAirbyteStream`](https://github.com/airbytehq/airbyte/blob/master/airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml#L158)

Reply:
I've changed the naming of local variables to make it super clear. You are right, `AirbyteStream` has `supported_sync_modes` and `ConfiguredAirbyteStream` has `sync_mode`. Here we are using a `ConfiguredAirbyteStream` object which has the `sync_mode` attribute.